src/Koffee/MainWindow.xaml

```diff
@@ -103,6 +103,38 @@
           </ItemsControl.ItemTemplate>
         </ItemsControl>
       </StackPanel>
+      <StackPanel Name="HistoryPanel" VerticalAlignment="Bottom" Visibility="Collapsed"
+                  Background="{DynamicResource PopupBrush}" Margin="32,0">
+        <TextBlock Text="History" Margin="8,4,8,8" Foreground="{DynamicResource PopupTextBrush}" FontWeight="Bold"/>
+        <ItemsControl Name="HistoryBack" Foreground="{DynamicResource PopupTextBrush}">
+          <ItemsControl.ItemTemplate>
+            <DataTemplate>
+              <StackPanel Orientation="Horizontal" Margin="8,0,8,6">
+                <TextBlock Text="{Binding Item1}" FontWeight="Bold" Width="30"/>
+                <TextBlock Text="{Binding Item2}"/>
+              </StackPanel>
+            </DataTemplate>
+          </ItemsControl.ItemTemplate>
+        </ItemsControl>
+        <ItemsControl Name="HistoryCurrent" Foreground="{DynamicResource PopupTextBrush}">
+          <StackPanel Orientation="Horizontal" Margin="8,0,8,6">
+            <TextBlock Text="-" FontWeight="Bold" Width="20" TextAlignment="Center"/>
```

Comment:
Should maybe not have a leader here? Maybe show a command to close the panel?

Ex:
- `1H` /some/path
- `gh` _(current)_ /some/path/2
- `1L` /some/path/2/foo

Or:
- `1H` /some/path
- `<Esc>` _(current)_ /some/path/2
- `1L` /some/path/2/foo

Reply:
Option 2 makes the most sense to me

Reply:
Was low-key hoping you were going to say that one as well. I also like that one the most :)